app/api/connection-details/route.ts

```diff
@@ -0,0 +1,61 @@
+import { ConnectionDetails } from '@/lib/types';
+import { randomUUID } from 'crypto';
+import { AccessToken, AccessTokenOptions, VideoGrant } from 'livekit-server-sdk';
+import { NextRequest, NextResponse } from 'next/server';
+
+const API_KEY = process.env.LIVEKIT_API_KEY;
+const API_SECRET = process.env.LIVEKIT_API_SECRET;
+const LIVEKIT_URL = process.env.LIVEKIT_URL;
+
+export async function GET(request: NextRequest) {
+  try {
+    // Parse query parameters
+    const roomName = request.nextUrl.searchParams.get('roomName');
+    const participantName = request.nextUrl.searchParams.get('participantName');
+    const metadata = request.nextUrl.searchParams.get('metadata') ?? '';
+
+    if (typeof roomName !== 'string') {
+      return new NextResponse('Missing required query parameter: roomName', { status: 400 });
+    }
+    if (participantName === null) {
+      return new NextResponse('Missing required query parameter: participantName', { status: 400 });
+    }
+
+    // Generate participant token
+    const participantToken = await createParticipantToken(
+      {
+        identity: `${participantName}__${randomUUID()}`,
+        name: participantName,
+        metadata,
+      },
+      roomName,
+    );
+
+    // Return connection details
+    const data: ConnectionDetails = {
+      server_url: LIVEKIT_URL!,
+      room_name: roomName ? roomName : 'playground_hello-world',
+      participant_token: participantToken,
+      participant_name: participantName ? participantName : 'playground-default-participant',
```

Comment:
same as above, I think we should require a participantName for meet

Reply:
Good catch! Hardcoded test stuff.